src/modules/firestore/index.js

```diff
@@ -180,6 +183,32 @@ export default class Firestore extends ModuleBase {
         )
       );
     }
+
+    if (hop(settings, 'cacheSizeBytes')) {
+      if (!isNumber(settings.cacheSizeBytes)) {
+        return Promise.reject(
+          new Error(
+            'Firestore.settings failed: settings.cacheSizeBytes must be number.'
+          )
+        );
+      }
+      if (settings.cacheSizeBytes < MIN_CACHE_SIZE) {
```

Comment:
Just a note that this will interact with the constant for disabling GC, as that's -1 however that is typed in the future might be good to put in now, but this is great for the "valid argument" check in general